src/module/system/action-macros/athletics/grapple.ts

```diff
@@ -51,10 +70,16 @@ function grapple(options: SkillActionOptions): void {
 }
 
 class GrappleActionVariant extends SingleCheckActionVariant {
-    protected override checkContext<ItemType extends ItemPF2e<ActorPF2e>>(
-        opts: CheckContextOptions<ItemType>,
-        data: CheckContextData<ItemType>,
-    ): CheckContext<ItemType> | undefined {
+    override async use(options: GrappleActionUseOptions = {}): Promise<CheckResultCallback[]> {
+        return super.use(
+            await ActionMacroHelpers.resolveWieldedWeaponFromItemParameter(options, REQUIRED_WEAPON_TRAIT),
```

Comment:
I don't think we need to do this. If we handle this resolving in the actual action implementatiion, we can avoid a separate interface type for "resolved" data, and reduce the number of layers of abstraction.